pumpkin/src/entity/mod.rs

```diff
@@ -47,6 +55,9 @@ pub struct Entity {
     /// Indicates whether the entity is on the ground (may not always be accurate).
     pub on_ground: AtomicBool,
 
+    /// Indicates whether the entity is inside of ground, and needs to be pushed out.
+    pub in_ground: AtomicBool,
```

Comment:
I assume this generally checks if a Entity is in blocks not only ground, right?, Then a better name would be maybe `inside_block`